### ISSUE TITLE
parserOptions.ecmaFeatures.jsx option not passed to estree 

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#24]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/594

---

**What version of TypeScript are you using?**
3.1.1

**What version of `typescript-eslint-parser` are you using?**
21.0.2

**What code were you trying to parse?**

```js
const RuleTester = require("eslint").RuleTester;

const ruleTester = new RuleTester({
    parserOptions: {
        ecmaVersion: 6,
        sourceType: "module",
        ecmaFeatures: {
            jsx: true,
        },
    },
    parser: "typescript-eslint-parser",
});

ruleTester.run("no-unused-vars", rule, {
    valid: [
		`
const Foo = function () {}
function render() {
    return (<Foo />);
}
		`,
	],
	invalid: [],
})
```

**What did you expect to happen?**
The code should parse successfully, so I can test the rule.

**What happened?**
`Parsing error: \'>\' expected.`

[see astexplorer repl](https://astexplorer.net/#/gist/45a09d9c2abdf3ae1d59e6b703d05913/d8b1c7667fe0549b931abbcc3078c8a665d5f8c4)
Note the parser has the `jsx` option set to true
